tools/export/cmake/__init__.py

```diff
@@ -0,0 +1,225 @@
+
+
+class CMake(Exporter):
+    """Generic CMake template that mimics the behavior of the python build
+    system
+    """
+
+    TEMPLATE = 'CMakeLists.txt'
+
+    MBED_CONFIG_HEADER_SUPPORTED = True
+
+    PREPROCESS_ASM = False
+
+    @classmethod
+    def is_target_supported(cls, target_name):
+        return True
+
+    def generate(self):
+        """Generate the CMakefiles.txt
+        """
+        self.resources.win_to_unix()
+
+        sources = set(self.resources.c_sources + \
+                      self.resources.cpp_sources + \
+                      self.resources.s_sources + \
+                      self.resources.headers)
+
+        libnames = [l[:-4] for l in self.resources.lib_refs]
+        libs = {re.sub(r'^[.]/', '', l): sorted([f for f in sources if f.startswith(l)]) for l in libnames}
+        libs = {k: v for k, v in libs.items() if len(v) != 0}
+        srcs = sorted([f for f in sources if f not in [item for sublist in libs.values() for item in sublist]])
+
+        libraries = [self.prepare_lib(basename(lib)) for lib
+                     in self.resources.libraries]
+        sys_libs = [self.prepare_sys_lib(lib) for lib
+                    in self.toolchain.sys_libs]
+
+        ctx = {
+            'name': self.project_name,
+            'target': self.target,
+            'sources': srcs,
+            'libs': libs,
+            'libraries': libraries,
+            'ld_sys_libs': sys_libs,
+            'include_paths': sorted(list(set(self.resources.inc_dirs))),
+            'library_paths': sorted(self.resources.lib_dirs),
+            'linker_script': self.resources.linker_script,
+            'hex_files': self.resources.hex_files,
+            'ar': basename(self.toolchain.ar),
+            'cc': basename(self.toolchain.cc[0]),
+            'cc_flags': " ".join(self.toolchain.cc[1:]),
+            'cxx': basename(self.toolchain.cppc[0]),
+            'cxx_flags': " ".join(self.toolchain.cppc[1:]),
+            'asm': basename(self.toolchain.asm[0]),
+            'asm_flags': " ".join(self.toolchain.asm[1:]),
+            'symbols': self.toolchain.get_symbols(),
+            'ld': basename(self.toolchain.ld[0]),
+            'ld_flags': " ".join(self.toolchain.ld[1:]),
+            'elf2bin': basename(self.toolchain.elf2bin),
+            'link_script_ext': self.toolchain.LINKER_EXT,
+            'link_script_option': self.LINK_SCRIPT_OPTION,
+            'user_library_flag': self.USER_LIBRARY_FLAG,
+            'needs_asm_preproc': self.PREPROCESS_ASM,
+        }
+
+        if hasattr(self.toolchain, "preproc"):
+            ctx['pp'] = " ".join(["\'" + part + "\'" for part
+                                  in ([basename(self.toolchain.preproc[0])] +
+                                      self.toolchain.preproc[1:] +
+                                      self.toolchain.ld[1:])])
+        else:
+            ctx['pp'] = None
+
+        for templatefile in ['cmake/%s.tmpl' % self.TEMPLATE]:
```

Comment:
Why are you looping through a list size one? Could you simplify this?

Reply:
I will take a look and update the PR. Some of your comments obviously relay to my lazyness...

Reply:
fixed